crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -728,11 +728,12 @@ pub trait Call:
         DB: Database + DatabaseCommit + OverrideBlockHashes,
         EthApiError: From<<DB as Database>::Error>,
     {
-        if request.as_ref().gas_limit() > Some(self.call_gas_limit()) {
-            // configured gas exceeds limit
-            return Err(
-                EthApiError::InvalidTransaction(RpcInvalidTransactionError::GasTooHigh).into()
-            )
+        if let Some(requested_gas) = request.as_ref().gas_limit() {
+            let global_gas_cap = self.call_gas_limit();
+            if global_gas_cap != 0 && global_gas_cap < requested_gas {
+                warn!(target: "rpc::eth::call", ?request, ?global_gas_cap, "Capping gas limit to global gas cap");
+                request.as_mut().set_gas_limit(global_gas_cap);
+            }
```

Comment:
I don't really like doing this because this assumes the user is fine with us altering their intended settings

but this is fine as long as configured gas cap is higher than the actual available blockspace